### ISSUE TITLE
Document valid forms of repository argument

### ISSUE THREAD

**Issue by bk2204** [actions/checkout#16]
I'm trying to set up GitHub Actions for CI on Git LFS.  We want to clone Git itself so that we can test against older versions, but it appears that I'm invoking things incorrectly.

I expected that the `repository` argument could be an arbitrary Git remote, and I've tried `https://github.com/git/git` and `https://github.com/git/git.git`, both of which are public, but neither appears to work.  The message I get is as follows:

```
  with:
    repository: https://github.com/git/git
    path: git
    ref: master
    clean: true
##[error]Specified argument was out of the range of valid values.
Parameter name: https://github.com/git/git
##[error]Exit code 1 returned from process: file name '/home/runner/runners/2.157.0/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Repository.CheckoutTask, Runner.Plugins"'.
```

If you're interested in an example run, the "Build with latest Git" target on https://github.com/git-lfs/git-lfs/runs/198689891 should be illustrative.

Could you document the valid forms of the repository argument so that folks can figure out how to use this functionality?

**Comment by TingluoHuang** [comment]
@bk2204 try
```
    repository: git/git
    path: git
    ref: refs/heads/master
    clean: true
```

**Comment by bk2204** [comment]
Okay, thanks for the info; that does appear to work.  Could you document that this only takes an NWO (maybe using another term than that) and can't be used to clone arbitrary remote URLs?  I feel like other people will be confused by this as well.

**Comment by ericsciple** [comment]
@bk2204 thanks, fixed with #87
